cli/azd/cmd/auth_token.go

```diff
@@ -47,6 +48,9 @@ func (f *authTokenFlags) Bind(local *pflag.FlagSet, global *internal.GlobalComma
 	f.global = global
 	local.StringArrayVar(&f.scopes, "scope", nil, "The scope to use when requesting an access token")
 	local.StringVar(&f.tenantID, "tenant-id", "", "The tenant id to use when requesting an access token.")
+	if exec.IsAzEmulator() {
```

Comment:
I didn't think about it, but it seems like a better approach.

Just to confirm, do you mean going all the way to the top `main.go` and condition the call to 

```
cmdErr := cmd.NewRootCmd(false, nil, rootContainer).ExecuteContext(ctx)
```

To have something like `cmd.NewAzEmulate(..)`  ???    Or would you also use a different package than `cmd` , like `azemulate.NewRootCmd` ?  so we would replace the entire `cmd` package?

Now that you mentioned this, I think it would need to disable the check for newer version code...

Reply:
> Now that you mentioned this, I think it would need to disable the check for newer version code...

That is slightly unfortunate, but I guess the switch would be happening at `main.go`, yeah...

Reply:
done. Moved it on top of pkg, as a sibling to cmd package. Looks much better